zzz.c

```diff
@@ -321,7 +328,7 @@ int main (int argc, char **argv) {
 	char *sleep_state = "mem";
 	char *hibernate_mode = "";
 
-	if (str_ends_with(argv[0], "/ZZZ")) {
+	if (path_ends_with(argv[0], "ZZZ")) {
```

Comment:
Wouldn’t this make the same job…?

```diff
- if (str_ends_with(argv[0], "/ZZZ")) {
+ if (strcmp(argv[0], "ZZZ") == 0 || str_ends_with(argv[0], "/ZZZ")) {
```

Reply:
Almost: It would do more work than necessary for the same result. It would call `strcmp` twice, comparing the string twice. That’s not important, though.

Far more important of a reason is that that would complicate control flow; make it harder to read: The logic of the comparison is already factored out into a function. Your proposed change would both retain the function (which is used only in this one place) *and* put logic into this condition.

While your change is much smaller—a one-line diff—it makes the resulting program harder to scan.

Two other alternatives are to do two `strcmp`s both in `main` or both in another function.

Reply:
That’s an interesting reasoning, but I disagree. `str_ends_with` is a general function, it’s present in almost every language’s stdlib and it’s very clear what it does; there’s no ambiguity or hidden assumptions. `path_ends_with` is much more complicated – what is a path? what does it mean that given path ends with a suffix? does `/` count? and so on.